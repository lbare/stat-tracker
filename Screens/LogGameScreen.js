import React, { useEffect, useState, useMemo } from "react";
import { View, Alert, TouchableOpacity, Text } from "react-native";
import StepIndicator from "react-native-step-indicator";
import HitLocation from "../components/HitLocation";
import Result from "../components/Result";
import Count from "../components/Count";
import Trajectory from "../components/Trajectory";
import Runs from "../components/Runs";

const LogGame = () => {
  const [activePage, setActivePage] = useState(0);
  const [result, setResult] = useState(null);
  const [count, setCount] = useState({
    balls: 0,
    strikes: 0,
  });
  const [hitLocation, setHitLocation] = useState({ x: 0, y: 0 });
  const [trajectory, setTrajectory] = useState(null);
  const [hardHit, setHardHit] = useState(null);
  const [RBI, setRBI] = useState(0);
  const [runScored, setRunScored] = useState(null);

  const canProceed = useMemo(() => {
    switch (activePage) {
      case 0:
        return result !== null;
      case 1:
        return true;
      case 2:
        return runScored !== null;
      case 3:
        return hitLocation.y !== 0;
      case 4:
        return trajectory !== null && hardHit !== null;
      default:
        return true;
    }
  }, [activePage, result, hitLocation.y, trajectory, hardHit, runScored]);

  const handleNext = () => setActivePage(activePage + 1);
  const handleBack = () => setActivePage(activePage - 1);

  const stepCount =
    result === "BB" || result === "K" || result === "HBP" ? 3 : 5;

  const content = {
    0: <Result result={result} setResult={setResult} />,
    1: <Count count={count} setCount={setCount} />,
    2: (
      <Runs
        runScored={runScored}
        setRunScored={setRunScored}
        RBI={RBI}
        setRBI={setRBI}
      />
    ),
    3: (
      <HitLocation hitLocation={hitLocation} setHitLocation={setHitLocation} />
    ),
    4: (
      <Trajectory
        trajectory={trajectory}
        setTrajectory={setTrajectory}
        hardHit={hardHit}
        setHardHit={setHardHit}
      />
    ),
  };

  const handleStep = (step) => {
    if (step === -1 && activePage === 0) return;
    else if (step === 1 && activePage === stepCount - 1) {
      Alert.alert("Confirm", "Are you sure you want to submit this at-bat?", [
        {
          text: "Cancel",
          style: "cancel",
        },
        {
          text: "OK",
          onPress: () => {
            console.log(
              stepCount === 5
                ? {
                    result: result,
                    hitLocation: {
                      x: Math.floor(hitLocation.x),
                      y: Math.floor(hitLocation.y),
                    },
                    count: count,
                    trajectory: trajectory,
                    hardHit: hardHit,
                    runScored: runScored,
                    RBI: RBI,
                  }
                : {
                    result: result,
                    count: count,
                    runScored: runScored,
                    RBI: RBI,
                  }
            );
          },
        },
      ]);
    } else setActivePage(activePage + step);
  };

  return (
    <View className="flex flex-col h-full w-full py-5 pb-48">
      <StepIndicator
        customStyles={{
          stepIndicatorSize: 25,
          currentStepIndicatorSize: 30,
          separatorStrokeWidth: 2,
          currentStepStrokeWidth: 3,
          stepStrokeCurrentColor: "#fe7013",
          stepStrokeWidth: 3,
          stepStrokeFinishedColor: "#fe7013",
          stepStrokeUnFinishedColor: "#aaaaaa",
          separatorFinishedColor: "#fe7013",
          separatorUnFinishedColor: "#aaaaaa",
          stepIndicatorFinishedColor: "#fe7013",
          stepIndicatorUnFinishedColor: "#ffffff",
          stepIndicatorCurrentColor: "#ffffff",
          stepIndicatorLabelFontSize: 13,
          currentStepIndicatorLabelFontSize: 13,
          stepIndicatorLabelCurrentColor: "#fe7013",
          stepIndicatorLabelFinishedColor: "#ffffff",
          stepIndicatorLabelUnFinishedColor: "#aaaaaa",
          labelColor: "#999999",
          labelSize: 13,
          currentStepLabelColor: "#fe7013",
        }}
        currentPosition={activePage}
        stepCount={stepCount}
      />
      <View
        className="flex-1 align-items justify-content"
        style={{ overflow: "hidden" }}
      >
        {content[activePage]}
      </View>

      <View className="flex flex-row justify-evenly">
        <TouchableOpacity
          className="bg-blue-500 w-1/3 h-16 justify-center items-center rounded-xl"
          onPress={() => handleStep(-1)}
        >
          <Text className="text-white text-2xl">Back</Text>
        </TouchableOpacity>

        <TouchableOpacity
          className="bg-blue-500 w-1/3 h-16 justify-center items-center rounded-xl"
          onPress={() => handleStep(1)}
          disabled={!canProceed}
        >
          <Text className="text-white text-2xl">
            {activePage === 4 ? "Finish" : "Next"}
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default LogGame;
