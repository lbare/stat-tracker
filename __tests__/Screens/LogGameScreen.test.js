import React from "react";
import renderer from "react-test-renderer";
import LogGameScreen from "../../Screens/LogGameScreen";

jest.mock("react-native-step-indicator");

describe("<LogGameScreen />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<LogGameScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
