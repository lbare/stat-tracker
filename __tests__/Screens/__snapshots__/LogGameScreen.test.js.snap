// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogGameScreen /> renders correctly 1`] = `
<View
  style={
    [
      {
        "display": "flex",
      },
      {
        "flexDirection": "column",
      },
      {
        "height": "100%",
      },
      {
        "width": "100%",
      },
      {
        "paddingBottom": 20,
        "paddingTop": 20,
      },
      {
        "paddingBottom": 192,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "overflow": "hidden",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          {
            "width": "100%",
          },
          {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 36,
              "lineHeight": 40,
            },
            {
              "fontWeight": "900",
            },
            {
              "textAlign": "center",
            },
            {
              "color": "#3b82f6",
            },
            {
              "paddingTop": 40,
            },
          ]
        }
      >
        Result
      </Text>
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "justifyContent": "space-evenly",
            },
            {
              "paddingBottom": 24,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 24,
            },
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#f3f4f6",
              "borderBottomColor": "#6b7280",
              "borderBottomLeftRadius": 12,
              "borderBottomRightRadius": 12,
              "borderBottomWidth": 1,
              "borderLeftColor": "#6b7280",
              "borderLeftWidth": 1,
              "borderRightColor": "#6b7280",
              "borderRightWidth": 1,
              "borderTopColor": "#6b7280",
              "borderTopLeftRadius": 12,
              "borderTopRightRadius": 12,
              "borderTopWidth": 1,
              "height": 56,
              "justifyContent": "center",
              "opacity": 1,
              "width": "20%",
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "center",
                },
                {
                  "fontSize": 20,
                  "lineHeight": 28,
                },
              ]
            }
          >
            1B
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#f3f4f6",
              "borderBottomColor": "#6b7280",
              "borderBottomLeftRadius": 12,
              "borderBottomRightRadius": 12,
              "borderBottomWidth": 1,
              "borderLeftColor": "#6b7280",
              "borderLeftWidth": 1,
              "borderRightColor": "#6b7280",
              "borderRightWidth": 1,
              "borderTopColor": "#6b7280",
              "borderTopLeftRadius": 12,
              "borderTopRightRadius": 12,
              "borderTopWidth": 1,
              "height": 56,
              "justifyContent": "center",
              "opacity": 1,
              "width": "20%",
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "center",
                },
                {
                  "fontSize": 20,
                  "lineHeight": 28,
                },
              ]
            }
          >
            2B
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#f3f4f6",
              "borderBottomColor": "#6b7280",
              "borderBottomLeftRadius": 12,
              "borderBottomRightRadius": 12,
              "borderBottomWidth": 1,
              "borderLeftColor": "#6b7280",
              "borderLeftWidth": 1,
              "borderRightColor": "#6b7280",
              "borderRightWidth": 1,
              "borderTopColor": "#6b7280",
              "borderTopLeftRadius": 12,
              "borderTopRightRadius": 12,
              "borderTopWidth": 1,
              "height": 56,
              "justifyContent": "center",
              "opacity": 1,
              "width": "20%",
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "center",
                },
                {
                  "fontSize": 20,
                  "lineHeight": 28,
                },
              ]
            }
          >
            3B
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#f3f4f6",
              "borderBottomColor": "#6b7280",
              "borderBottomLeftRadius": 12,
              "borderBottomRightRadius": 12,
              "borderBottomWidth": 1,
              "borderLeftColor": "#6b7280",
              "borderLeftWidth": 1,
              "borderRightColor": "#6b7280",
              "borderRightWidth": 1,
              "borderTopColor": "#6b7280",
              "borderTopLeftRadius": 12,
              "borderTopRightRadius": 12,
              "borderTopWidth": 1,
              "height": 56,
              "justifyContent": "center",
              "opacity": 1,
              "width": "20%",
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "center",
                },
                {
                  "fontSize": 20,
                  "lineHeight": 28,
                },
              ]
            }
          >
            HR
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "justifyContent": "space-evenly",
            },
            {
              "paddingLeft": 24,
              "paddingRight": 24,
            },
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#f3f4f6",
              "borderBottomColor": "#6b7280",
              "borderBottomLeftRadius": 12,
              "borderBottomRightRadius": 12,
              "borderBottomWidth": 1,
              "borderLeftColor": "#6b7280",
              "borderLeftWidth": 1,
              "borderRightColor": "#6b7280",
              "borderRightWidth": 1,
              "borderTopColor": "#6b7280",
              "borderTopLeftRadius": 12,
              "borderTopRightRadius": 12,
              "borderTopWidth": 1,
              "height": 56,
              "justifyContent": "center",
              "opacity": 1,
              "width": "20%",
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "center",
                },
                {
                  "fontSize": 20,
                  "lineHeight": 28,
                },
              ]
            }
          >
            HBP
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#f3f4f6",
              "borderBottomColor": "#6b7280",
              "borderBottomLeftRadius": 12,
              "borderBottomRightRadius": 12,
              "borderBottomWidth": 1,
              "borderLeftColor": "#6b7280",
              "borderLeftWidth": 1,
              "borderRightColor": "#6b7280",
              "borderRightWidth": 1,
              "borderTopColor": "#6b7280",
              "borderTopLeftRadius": 12,
              "borderTopRightRadius": 12,
              "borderTopWidth": 1,
              "height": 56,
              "justifyContent": "center",
              "opacity": 1,
              "width": "20%",
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "center",
                },
                {
                  "fontSize": 20,
                  "lineHeight": 28,
                },
              ]
            }
          >
            SF
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#f3f4f6",
              "borderBottomColor": "#6b7280",
              "borderBottomLeftRadius": 12,
              "borderBottomRightRadius": 12,
              "borderBottomWidth": 1,
              "borderLeftColor": "#6b7280",
              "borderLeftWidth": 1,
              "borderRightColor": "#6b7280",
              "borderRightWidth": 1,
              "borderTopColor": "#6b7280",
              "borderTopLeftRadius": 12,
              "borderTopRightRadius": 12,
              "borderTopWidth": 1,
              "height": 56,
              "justifyContent": "center",
              "opacity": 1,
              "width": "20%",
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "center",
                },
                {
                  "fontSize": 20,
                  "lineHeight": 28,
                },
              ]
            }
          >
            SAC
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#f3f4f6",
              "borderBottomColor": "#6b7280",
              "borderBottomLeftRadius": 12,
              "borderBottomRightRadius": 12,
              "borderBottomWidth": 1,
              "borderLeftColor": "#6b7280",
              "borderLeftWidth": 1,
              "borderRightColor": "#6b7280",
              "borderRightWidth": 1,
              "borderTopColor": "#6b7280",
              "borderTopLeftRadius": 12,
              "borderTopRightRadius": 12,
              "borderTopWidth": 1,
              "height": 56,
              "justifyContent": "center",
              "opacity": 1,
              "width": "20%",
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "center",
                },
                {
                  "fontSize": 20,
                  "lineHeight": 28,
                },
              ]
            }
          >
            E
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "justifyContent": "space-evenly",
            },
            {
              "paddingLeft": 40,
              "paddingRight": 40,
            },
            {
              "paddingTop": 24,
            },
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#f3f4f6",
              "borderBottomColor": "#6b7280",
              "borderBottomLeftRadius": 12,
              "borderBottomRightRadius": 12,
              "borderBottomWidth": 1,
              "borderLeftColor": "#6b7280",
              "borderLeftWidth": 1,
              "borderRightColor": "#6b7280",
              "borderRightWidth": 1,
              "borderTopColor": "#6b7280",
              "borderTopLeftRadius": 12,
              "borderTopRightRadius": 12,
              "borderTopWidth": 1,
              "height": 56,
              "justifyContent": "center",
              "opacity": 1,
              "width": "20%",
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "center",
                },
                {
                  "fontSize": 20,
                  "lineHeight": 28,
                },
              ]
            }
          >
            BB
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#f3f4f6",
              "borderBottomColor": "#6b7280",
              "borderBottomLeftRadius": 12,
              "borderBottomRightRadius": 12,
              "borderBottomWidth": 1,
              "borderLeftColor": "#6b7280",
              "borderLeftWidth": 1,
              "borderRightColor": "#6b7280",
              "borderRightWidth": 1,
              "borderTopColor": "#6b7280",
              "borderTopLeftRadius": 12,
              "borderTopRightRadius": 12,
              "borderTopWidth": 1,
              "height": 56,
              "justifyContent": "center",
              "opacity": 1,
              "width": "20%",
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "center",
                },
                {
                  "fontSize": 20,
                  "lineHeight": 28,
                },
              ]
            }
          >
            K
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#f3f4f6",
              "borderBottomColor": "#6b7280",
              "borderBottomLeftRadius": 12,
              "borderBottomRightRadius": 12,
              "borderBottomWidth": 1,
              "borderLeftColor": "#6b7280",
              "borderLeftWidth": 1,
              "borderRightColor": "#6b7280",
              "borderRightWidth": 1,
              "borderTopColor": "#6b7280",
              "borderTopLeftRadius": 12,
              "borderTopRightRadius": 12,
              "borderTopWidth": 1,
              "height": 56,
              "justifyContent": "center",
              "opacity": 1,
              "width": "20%",
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "center",
                },
                {
                  "fontSize": 20,
                  "lineHeight": 28,
                },
              ]
            }
          >
            OUT
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "display": "flex",
        },
        {
          "flexDirection": "row",
        },
        {
          "justifyContent": "space-evenly",
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#3b82f6",
          "borderBottomLeftRadius": 12,
          "borderBottomRightRadius": 12,
          "borderTopLeftRadius": 12,
          "borderTopRightRadius": 12,
          "height": 64,
          "justifyContent": "center",
          "opacity": 1,
          "width": "33.333333%",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#fff",
            },
            {
              "fontSize": 24,
              "lineHeight": 32,
            },
          ]
        }
      >
        Back
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#3b82f6",
          "borderBottomLeftRadius": 12,
          "borderBottomRightRadius": 12,
          "borderTopLeftRadius": 12,
          "borderTopRightRadius": 12,
          "height": 64,
          "justifyContent": "center",
          "opacity": 1,
          "width": "33.333333%",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#fff",
            },
            {
              "fontSize": 24,
              "lineHeight": 32,
            },
          ]
        }
      >
        Next
      </Text>
    </View>
  </View>
</View>
`;
